programs/account-compression/src/instructions/append_leaves.rs

```diff
@@ -111,16 +126,34 @@ pub fn process_append_leaves_to_merkle_trees<'a, 'b, 'c: 'info, 'info>(
             paths.push(path);
         }
 
-        #[cfg(target_os = "solana")]
-        light_heap::GLOBAL_ALLOCATOR.log_total_heap("append_leaves: after changelog event");
-
         changelog_events.push(ChangelogEvent::V1(ChangelogEventV1 {
             id: merkle_tree_pubkey.to_bytes(),
             paths,
             seq: first_sequence_number as u64,
             index: merkle_tree.changelog[first_changelog_index].index as u32,
         }));
+
+        leaves_start += leaves_end;
+
+        if leaves_start == BATCH_SIZE {
```

Comment:
what happens if we don't have enough leaves to fill the last batch completely?

Reply:
The loop will exit. But good point, we should detect such case outside the loop and emit the event

Reply:
Or no, actually this is all wrong. I should rather compare the incremented `leaves_start` with `min::(leaves.len(), BATCH_SIZE)`